main/src/main/scala/sbt/Keys.scala

```diff
@@ -222,11 +237,13 @@ object Keys {
   val consoleProject = taskKey[Unit]("Starts the Scala interpreter with the sbt and the build definition on the classpath and useful imports.").withRank(AMinusTask)
   val compile = taskKey[CompileAnalysis]("Compiles sources.").withRank(APlusTask)
   val manipulateBytecode = taskKey[CompileResult]("Manipulates generated bytecode").withRank(BTask)
-  val compileIncremental = taskKey[CompileResult]("Actually runs the incremental compilation").withRank(DTask)
+  val compileIncremental = taskKey[(Boolean, HashedVirtualFileRef)]("Actually runs the incremental compilation").withRank(DTask)
```

Comment:
iirc, I had to weaken `CompileResult` to `(Boolean, HashedVirtualFileRef)` because `CompileResult` contains `File` or something, and wanted to avoid it. `CompileResult` contains `hasModified`.
This is used in sbt 1.x for pipelining etc - https://github.com/sbt/sbt/blob/v1.10.0-M1/main/src/main/scala/sbt/Defaults.scala#L2260. We might have to revisit this area, potentially by making `CompileResult` machine-independent.